src/client/world.ts

```diff
@@ -268,6 +286,16 @@ class World {
     return null;
   }
 
+  fetchImprovementCatalogs(improvement: Improvement, coords: Coords): void {
+    if (improvement.type === 'encampment') {
+      // change this check later, to be more general
+      this.sendActions([[ 'getUnitCatalog', [coords] ]])
+    } else if (improvement.type === 'campus') {
+      // change this check later, to be more general
+      this.sendActions([[ 'getKnowledgeCatalog', [coords] ]])
```

Comment:
Correct. Since for some reason we've decided to only use callbacks in the frontend (again, we should probably eventually just simply build a new frontend) we kinda have to do it this way.